pkg/disttask/framework/taskexecutor/manager.go

```diff
@@ -322,72 +309,24 @@ func (m *Manager) handleExecutableTask(task *proto.Task) {
 		return
 	}
 	executor := factory(m.ctx, m.id, task, m.taskTable)
-	taskCtx, taskCancel := context.WithCancelCause(m.ctx)
-	m.registerCancelFunc(task.ID, taskCancel)
-	defer taskCancel(nil)
-	// executor should init before run()/pause()/rollback().
-	err := executor.Init(taskCtx)
+	err := executor.Init(m.ctx)
```

Comment:
If we use `m.ctx` here, we may encounter the same problem as #48343?

> the context passed to the local backend is wrong. It should be "task context" instead of "manager context", because the latter will only be canceled when the TiDB process exits.

If `TaskExecutor.Init()` is designed to use `m.ctx`, Maybe we should move backend ctx initialization to `stepExecutor.Init`.

Reply:
> Maybe we should move backend ctx initialization to stepExecutor.Init.

yes, local backend should be started where it's needed(when run a step), and `step run context` should be used.

Reply:
add a temp fix and TODO
https://github.com/pingcap/tidb/pull/50724/commits/34cc3e231a2e4334aeda9185b7ee75b76b017cd1